### PR TITLE
Maximum Topology Short in C++

### PR DESCRIPTION
**Pull request by codeakki**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Maximum Topology Short in C++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4389

### CODE REVIEW

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing due to inactivity.

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -111,6 +111,7 @@
 - [First Negative Integer in Every k sized window](cp/1st_negative_no_in_every_window_of_size_k.cpp)
 - [Rotating an NxN 2D array in 90 degrees anticlockwise direction](cp/rotate2darray.cpp)
 - [Count occurence of Anagram](cp/count_occurences_of_anagrams.cpp)
+- [Maximum Topology Short ] (cp/Max_Topology_short.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md:114]: Fix the hyperlink here

File: C-Plus-Plus/cp/Max_Topology_short.cpp [hunk 2]
```diff
@@ -0,0 +1,207 @@
+/*Maximum Topology Short 
+For example, consider the tree below:
+              2
+        1           3 
+
+
+The sorted list of pairs is ((2,2),(1,3),(1,1))
+*/
+/*To Include All Pre Required Liabries*/
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Max_Topology_short.cpp:9]: Give a description of the algorithm. Share the examples at the bottom

File: C-Plus-Plus/cp/Max_Topology_short.cpp [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    }
+}
+/* Example input
+
+4 2
+1 2
+2 3
+2 4 
+Output:
+4 2 */
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Max_Topology_short.cpp:207]: Add a new line at the EOF

File: C-Plus-Plus/cp/Max_Topology_short.cpp [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+#define int long long int
+#define mod 1000000007 
+
+
+int n;
+const int Limit = (int)5e5 + 5;
+int fni[Limit + 1];
+/*Fni-->Factorail Number inverse*/
+int nni[Limit + 1];
+int f_a_c_t[Limit + 1];
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Max_Topology_short.cpp:21]: Use meaningful variable names. Not just here bu throughout the code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Found same PR opened by you. Closing this as of now.

**Pull request closed without merging**